### DISCUSSION TITLE
Typescript gives error when import schemas from different file.

### DISCUSSION THREAD

**Discussion by yusufilker-mnm** [keystonejs/keystone#9428] [Questions] [answered]
“I created a new project using the command `npm create keystone-app@latest`. Then, I tried to move the User list from the `schema.ts` file to a different file within the project. However, when I did this, it caused a TypeScript type error.”

The Error: `“Type ‘{ isSingleton?: boolean | undefined; fields: BaseFields; access: ListAccessControl; … 6 more …; defaultIsOrderable?: boolean | … 1 more … | undefined; }’ is not assignable to type ‘ListConfig<TypeInfo>’.
The types of the ‘fields’ property are incompatible.
‘BaseFields’ is not assignable to ‘BaseFields<TypeInfo>’.
The ‘string’ index signatures are incompatible.
‘FieldTypeFunc’ is not assignable to ‘FieldTypeFunc<TypeInfo>’.
‘NextFieldType<DBField, Arg | undefined, Arg, Arg<NullableInputType, false>, Arg<NullableInputType, false>, Arg<…>, BaseListTypeInfo>’ is not assignable to ‘NextFieldType<DBField, Arg | undefined, Arg, Arg<NullableInputType, false>, Arg<NullableInputType, false>, Arg<…>, TypeInfo<…>>’.
‘NextFieldType<DBField, Arg | undefined, Arg, Arg<NullableInputType, false>, Arg<NullableInputType, false>, Arg<…>, BaseListTypeInfo>’ is not assignable to ‘CommonFieldConfig<TypeInfo>’.
The types of the ‘access’ property are incompatible.
‘FieldAccessControl | undefined’ is not assignable to ‘FieldAccessControl<TypeInfo> | undefined’.
‘{ read?: IndividualFieldAccessControl<FieldReadItemAccessArgs> | undefined; create?: IndividualFieldAccessControl<…> | undefined; update?: IndividualFieldAccessControl<…> | undefined; }’ is not assignable to ‘FieldAccessControl<TypeInfo> | undefined’.
‘{ read?: IndividualFieldAccessControl<FieldReadItemAccessArgs> | undefined; create?: IndividualFieldAccessControl<…> | undefined; update?: IndividualFieldAccessControl<…> | undefined; }’ is not assignable to ‘{ read?: IndividualFieldAccessControl<FieldReadItemAccessArgs<TypeInfo>> | undefined; create?: IndividualFieldAccessControl<FieldCreateItemAccessArgs<…>> | undefined; update?: IndividualFieldAccessControl<…> | undefined; }’.
The types of the ‘read’ property are incompatible.
‘IndividualFieldAccessControl<FieldReadItemAccessArgs> | undefined’ is not assignable to ‘IndividualFieldAccessControl<FieldReadItemAccessArgs<TypeInfo>> | undefined’.
‘IndividualFieldAccessControl<FieldReadItemAccessArgs>’ is not assignable to ‘IndividualFieldAccessControl<FieldReadItemAccessArgs<TypeInfo>>’.
‘FieldReadItemAccessArgs<TypeInfo>’ is not assignable to ‘FieldReadItemAccessArgs’.
‘FieldReadItemAccessArgs<TypeInfo>’ is not assignable to ‘BaseAccessArgs’.
The ‘context.db’ types are incompatible.
‘KeystoneDbAPI<{ readonly User: TypeInfo; readonly Post: TypeInfo; readonly Tag: TypeInfo; }>’ is not assignable to ‘KeystoneDbAPI<Record<string, BaseListTypeInfo>>’.
The ‘User’ property is incompatible with the index signature.
‘DbAPI<TypeInfo>’ is not assignable to ‘DbAPI<BaseListTypeInfo>’.
‘string’ is not assignable to ‘“User”’.ts(2322)
types.ts(369, 14): The expected type comes from the ‘User’ property declared in the ‘Lists’ type.”`

![Ekran Resmi 2024-12-03 14 20 48](https://github.com/user-attachments/assets/8d5d0ed6-866e-4f45-8bce-50159c43f361)
![Ekran Resmi 2024-12-03 13 45 17](https://github.com/user-attachments/assets/d8156ff9-4377-400a-93a3-b27edcb15eee)
![Ekran Resmi 2024-12-03 14 18 22](https://github.com/user-attachments/assets/9775a04a-6e33-415e-bcbc-e23b9a776a40)

**Comment by sam-gab** [comment, marked as answer]
In your `User.ts` file, add `satisfies Lists.User` to the end.  

So it will look like :
`export const User = list({ ..... } satisfies Lists.User`

Import `Lists` from `import { Lists } from ".keystone/types";`

**Comment by yusufilker-mnm** [comment]
Yeah it is worked. Thanks @sam-gab
